### PR TITLE
Support special characters in test case names

### LINKED ISSUE #25: Support special characters in test case names
Support special characters, e.g., a comma, in test case names. They must be excluded when building item IDs.

### PR DESCRIPTION
**Pull request by JasperCraeghs**
Closes #25

### CODE REVIEW

File: doc/source/robot/example.robot [hunk 1]
```diff
@@ -9,7 +9,7 @@ ${MESSAGE}     Hello,
 ...    world!
 
 *** Test Cases ***
-First Test
+First, Test
```
Comment (Letme) [inline doc/source/robot/example.robot]: I would rather have specific faulty test cases, just to keep the example more clean. This now implies that you can add commas to the test case title, but I do not want to make that as a standard assumption.

**Review by Letme — CHANGES_REQUESTED** [review]

File: doc/source/robot/example.robot (hunk 1 above)
Comment (JasperCraeghs) [reply doc/source/robot/example.robot]: 937d6020767d6a6c368880262a694f10866c75ca

File: doc/source/robot/example.robot [hunk 2]
```diff
@@ -20,7 +20,7 @@ First Test
 Undocumented Test
                         Should Be Equal     ${MESSAGE}    Hello, world!
 
-Test with documentation in RST syntax
+Test with documentation (in "RST" syntax)
```
Comment (Letme) [inline doc/source/robot/example.robot]:
```suggestion
Test with documentation in RST syntax
```

File: doc/source/robot/example.robot [hunk 3]
```diff
@@ -37,6 +37,10 @@ Another Test
                         FOR     ${var}  IN  @{NAMES}
                                 Log     ${var}
 
+Comp1: testing 'Special" characters & prefix
```
Comment (Letme) [inline doc/source/robot/example.robot]:
```suggestion
Comp1: testing 'Special" characters & prefix (with brackets)
```

**Review by Letme — CHANGES_REQUESTED** [review]

File: doc/source/robot/example.robot (hunk 3 above)
Comment (JasperCraeghs) [reply doc/source/robot/example.robot]: 5c4ffa5aa650531c728df6d89a87b98192531556

File: doc/source/robot/example.robot (hunk 2 above)
Comment (JasperCraeghs) [reply doc/source/robot/example.robot]: 5c4ffa5aa650531c728df6d89a87b98192531556

**Review by Letme — APPROVED** [review]

**Pull request merged**